### PR TITLE
Trimmomatic

### PR DESCRIPTION
**Pull request by alyssa-ab**
## PR checklist

Issue #1720 

- [X] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [X] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [X] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [X] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/trimmomatic/main.nf [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    """
+    trimmomatic \\
+        --cores $task.cpus \\
+        $args \\
+        $trimmed \\
+        > ${prefix}.trimmomatic.log
+        
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        trimmomatic: \$(trimmomatic --version)
```
Comment (Emiller88) [inline modules/trimmomatic/main.nf:34]:
```suggestion
        trimmomatic: \$(trimmomatic -version)
```

File: tests/modules/trimmomatic/test.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+## TODO nf-core: Please run the following command to build this file:
+#                nf-core modules create-test-yml trimmomatic
+- name: "trimmomatic"
+  command: nextflow run ./tests/modules/trimmomatic -entry test_trimmomatic -c ./tests/config/nextflow.config -c ./tests/modules/trimmomatic/nextflow.config
```
Comment (Emiller88) [inline tests/modules/trimmomatic/test.yml:4]:
```suggestion
  command: nextflow run ./tests/modules/trimmomatic -entry test_trimmomatic_single_end -c ./tests/config/nextflow.config -c ./tests/modules/trimmomatic/nextflow.config
```

**Review by Emiller88 — CHANGES_REQUESTED** [review]

**Review by Emiller88 — APPROVED** [review]
Looks good to me!

**Review by Emiller88 — CHANGES_REQUESTED** [review]
Just kidding, the files need to be added to `tests/config/pytest_modules.yml`!

**Review by Emiller88 — APPROVED** [review]
LGTM, I want someone else to check it as well since I helped write it.

File: modules/trimmomatic/main.nf [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def trimmed = meta.single_end ? "SE" : "PE"
+    def output = meta.single_end ?
+        "${prefix}.SE.paired.trim.fastq.gz" // HACK to avoid unpaired and paired in the trimmed_reads output
+        : "${prefix}.paired.trim_1.fastq.gz ${prefix}.unpaired.trim_1.fastq.gz ${prefix}.paired.trim_2.fastq.gz ${prefix}.unpaired.trim_2.fastq.gz"
+    // TODO Give better error output
```
Comment (JoseEspinosa) [inline modules/trimmomatic/main.nf:29]: Is this something you are leaving for a new iteration or just missed it on this one?

**Review by JoseEspinosa — APPROVED** [review]
LGTM! Just made a small question. 🚀

File: modules/trimmomatic/main.nf (hunk 3 above)
Comment (Emiller88) [reply modules/trimmomatic/main.nf:29]: Yeah, it works :tm: for now. It's not perfect naming, but trimmomatic is so esoteric with its cli. This is @alyssa-ab's first module and didn't want to spend even more time making it perfect the first go around.

So in conclusion, open to suggestions if anyone has them, but in a new PR.

**Pull request merged**